docs/operations_tasks/Ingest-Release-SOP.md

```diff
@@ -8,73 +8,80 @@ parent: Operations tasks
+    6. Track the deployment in the CI/CD page
+13. Ensure your changes are live on prod
+    - You may used the `kubectl get pods` command to check this
+    - Consult with [these docs](https://github.com/ebi-ait/ingest-kube-deployment) for further information on setting up `kubectl`
+
+### Manually deploying without GitLab
+You can manually deploy using the `ingest-kube-deployment` repo if GitLab is down and the deployment is urgent. This should only be done in an emergency and GitLab should be fixed preferentially.
+
+**Deploying manually will disable all of the GitLab tracking features so that the Operations > Environments page is no longer functional. A subsequent deployment via GitLab will return things to the normal state.**
+
+1. Tag the commit you want to deploy with the [git release tool](https://github.com/rdgoite/hca-developer-tools/blob/master/gitconfig).
+    - [quay.io](quay.io) will automatically build tagged commits
+2. Find the tag of the built image.
+    - You will need the full URL e.g. `quay.io/ebi-ait/ingest-core:123abc`
+    - Use this in the next step
+3. Follow the steps [here](https://github.com/ebi-ait/ingest-kube-deployment#manually-deploy-one-kubernetes-dockerized-applications-to-an-environment-aws) to deploy the built image.
+
+### Details on GitLab pipeline
+#### Shared config
+All of the projects use [this template](https://github.com/ebi-ait/gitlab-ci-templates/blob/master/build-release-deploy.yml) for their CI pipelines. This template can be customized on a per-project basis if neccesary. In most cases, the only addition needed is to override the `Unit Test` stage to run the project's unit tests.
+
+#### Stages
+```
+<feature branch> Unit Test -> Build
+
+<dev branch> Unit Test -> Build -> Release dev -> Deploy dev -> Integration dev
+
+<master branch> Unit Test -> Build -> Release staging -> Deploy staging -> Integration staging
+                                      Release prod (manual) -> Deploy prod (manual) -> Integration prod
+```
+
+Unit tests and build steps are ran on a feature branch when there is an open PR for that branch. A deployment to staging will be done for every commit to master and a deployment to prod must be triggered manually via the GitLab UI.
+
+#### Image releases
+The CI/CD pipeline publishes images to [quay.io](https://quay.io/repository/ebi-ait). Images are first tagged with the shortened commit hash and then retagged as `<environment>-<short commit hash>-<date>.<timestamp>` (e.g. `prod-33242e93-16-04-2021.1618577888`) for a release. This tag is what is deployed to the environments.
+
+#### Kubernetes deployments
+GitLab is integrated with the kubernetes cluster and you can see the status of the clusters and configure them [here](https://gitlab.ebi.ac.uk/hca/ingest-staging-manager/-/clusters).
+
+Deployments are done in the pipeline by cloning the `ingest-kube-deployment` repo, sourcing the config files to get the replica counts, and deploying via helm.
```

Comment:
Here we have a few shell compatibility issues, don't we? If so, please point them out here.

Reply:
Those issues are well documented at https://github.com/ebi-ait/ingest-kube-deployment/tree/pop+zsh. I don't think duplicating this will help as it increases the chance of things going out of date

Reply:
Besides, the shell compatibility issues are not relevant to GitLab